src/components/shared/chart/Canvas.js

```diff
@@ -151,6 +157,19 @@ export default class ChartCanvas<HoveredItem> extends React.Component<
     this._canvas = canvas;
   }
 
+  componentWillReceiveProps() {
+    const { hoveredItem, x, y } = this.state;
+    if (hoveredItem !== null) {
+      // It is possible that the data backing the chart has been
+      // changed, for instance after symbolication. To be sure the
+      // hovered item we've stored isn't invalid, re-set it to
+      // whatever is below the cursor now.
+      this.setState({
+        hoveredItem: this.props.hitTest(x, y),
```

Comment:
probably not a big deal, but for correctness I think we should use `nextProps.hitTest` (`nextProps` is passed to this function).

Reply:
Also, this doesn't seem to work for me: for some reason, hoveredItem is always null for me here... So the tooltip just disappears -- which is already better than the previous behavior!

I tried to look closer.
When recalculating `hitTest` here, we have:
* viewportTop: 0
* containerWidth: 0

while normally:
* viewportTop: 1990
* containerWidth: 1024

Can you try to find out why this is happening?

Reply:
Nice find!

It is `componentWillReceiveProps` in Viewport.js that sets the values briefly to zero:

```javascript
componentWillReceiveProps(newProps: ViewportProps) {
  if (
    this.props.viewportProps.viewportNeedsUpdate(/*...*/)
  ) {
    this.setState(this.getDefaultState(newProps));
    this._setSizeNextFrame();
  } else if /* ... */
}
```

The first line resets the fields to 0, and the second line causes the values to be recalculated in the next animation frame. Why it does so I don't know. I get two corresponding calls to `ChartCanvas:componentWillReceiveProps`, where `hitTest` first returns `null` and the second time it may return a valid item.

I think the simplest solution is to just assign `null` to `hoveredItem` unconditionally. The user just have to move the mouse slightly to make a correct tooltip appear again.